src/hotspot/share/ci/ciTypeFlow.hpp

```diff
@@ -342,7 +342,7 @@ class ciTypeFlow : public ResourceObj {
       ciType* array = pop_value();
       if (array == null_type())  return NULL;
       // Value type arrays may contain oop or flattened representation
-      assert(array->is_obj_array_klass() || (ValueArrayFlatten && array->is_value_array_klass()),
+      assert(array->is_obj_array_klass() || (FlatArrayFlatten && array->is_value_array_klass()),
```

Comment:
Shouldn't is_value_array_klass() be is_flat_array_klass()?

Reply:
I think is_value_array_klass() is correct because array is a ciType*, and I did not rename ci related functions.